api/lib/application/users/index.js

```diff
@@ -262,6 +262,46 @@ exports.register = async function(server) {
         tags: ['api', 'user'],
       },
     },
+    {
+      method: 'POST',
+      path: '/api/users/{id}/update-email',
+      config: {
+        pre: [
+          {
+            method: securityPreHandlers.checkRequestedUserIsAuthenticatedUser,
+            assign: 'requestedUserIsAuthenticatedUser',
+          },
+          {
+            method: featureToggles.checkIfEmailValidationIsEnabled,
+            assign: 'isEmailValidationEnabled',
+          },
+        ],
+        handler: userController.updateUserEmailWithValidation,
+        validate: {
+          params: Joi.object({
+            id: identifiersType.userId,
+          }),
+          options: {
+            allowUnknown: true,
+          },
+          payload: Joi.object({
+            data: {
+              type: Joi.string().valid('users').required(),
+              attributes: {
+                code: Joi.string().regex(/^[0-9]{6}$/).required(),
```

Comment:
le 0 n'est pas permis
```suggestion
                code: Joi.string().regex(/^[1-9]{6}$/).required(),
```

Reply:
Merci :) je corrige